cloudflare.go

```diff
@@ -109,22 +109,22 @@ func (api *API) makeRequestWithAuthType(method, uri string, params interface{},
 		return nil, errors.Wrap(err, "could not read response body")
 	}
 
-	switch resp.StatusCode {
-	case http.StatusOK:
-		break
-	case http.StatusUnauthorized:
-		return nil, errors.Errorf("HTTP status %d: invalid credentials", resp.StatusCode)
-	case http.StatusForbidden:
-		return nil, errors.Errorf("HTTP status %d: insufficient permissions", resp.StatusCode)
-	case http.StatusServiceUnavailable, http.StatusBadGateway, http.StatusGatewayTimeout,
-		522, 523, 524:
-		return nil, errors.Errorf("HTTP status %d: service failure", resp.StatusCode)
-	default:
-		var s string
-		if body != nil {
-			s = string(body)
+	if resp.StatusCode < http.StatusOK || resp.StatusCode > http.StatusIMUsed {
```

Comment:
Alternatively, I can do a true switch:

```go
switch {
	case resp.StatusCode >= http.StatusOK && resp.StatusCode <= http.StatusIMUsed:

Reply:
The switch would be preferred, if you can.